src/Operations.jl

```diff
@@ -337,7 +337,11 @@ function resolve_versions!(env::EnvCache, registries::Vector{Registry.RegistryIn
     for pkg in pkgs
         names[pkg.uuid] = pkg.name
     end
-    reqs = Resolve.Requires(pkg.uuid => VersionSpec(pkg.version) for pkg in pkgs)
+
+    # Unless using the unbounded or historical resolver, always allow stdlibs to update. Helps if the previous resolve
+    # happened on a different julia version / commit and the stdlib version in the manifest is not the current stdlib version
+    unbind_stdlibs = julia_version === VERSION
+    reqs = Resolve.Requires(pkg.uuid => is_stdlib(pkg.uuid) && unbind_stdlibs ? VersionSpec("*") : VersionSpec(pkg.version) for pkg in pkgs)
```

Comment:
@staticfloat it just occurred to me, should it be this instead? Maybe `VersionSpec("*")` would've allowed updating to latest always, beyond what's allowed for the given julia version?
```suggestion
    reqs = Resolve.Requires(pkg.uuid => is_stdlib(pkg.uuid) && unbind_stdlibs ? VersionSpec(stdlibs()[uuid][2]) : VersionSpec(pkg.version) for pkg in pkgs)
```

Reply:
I think in practice it doesn't matter, because as long as the compat ranges that we declare here match what is already in the system image, the resolver should choose those versions, right?  The issue isn't that we're trying to install a different version than exists, it's that the resolver isn't being allowed to use the versions that are already installed, I thought?

Reply:
Sounds reasonable to me. I'll merge